install.js

```diff
@@ -1,63 +1,89 @@
-
-    proc.execSync('git submodule update --init');
-    if (!checkLibuiohookSource()) {
-      deleteFolderRecursive(path.join(__dirname, 'libuiohook'));
-      proc.execSync('git clone https://github.com/kwhat/libuiohook.git');
-    }
-    
-    proc.execSync('npm install nan cmake-js');
-    proc.execSync('npm run compile');
-    // if (fs.existsSync(path.join(__dirname, 'build', 'Release', 'iohook.node'))) {
-    //   fs.renameSync(
-    //     path.join(__dirname, 'build', 'Release', 'iohook.node'),
-    //     path.join(__dirname, 'build', 'Release', currentPlatform + '.node')
-    //   );
-    // }
-  }
-} else {
-  proc.execSync('npm remove nan cmake-js');
-  deleteFolderRecursive(path.join(__dirname, 'libuiohook'), true);
-  deleteFolderRecursive(path.join(__dirname, 'build', 'CMakeFiles'));
+function onerror(err) {
+  console.error(err)
 }
 
-function checkLibuiohookSource() {
-  return fs.existsSync(path.join(__dirname, 'libuiohook', 'include', 'uiohook.h'));
+function install(runtime, abi, platform, arch, cb) {
+  const essential = runtime + '-v' + abi + '-' + platform + '-' + arch
+  const pkgVersion = pkg.version
+  const currentPlatform = pkg.name + '-v' + pkgVersion + '-' + essential
+  console.log('Downloading prebuild for platform:', currentPlatform);
+  const downloadUrl = 'https://github.com/vespakoen/iohook/releases/download/v' + pkgVersion + '/' + currentPlatform + '.tar.gz'
+  var reqOpts = { url: downloadUrl }
+  var tempFile = path.join(os.tmpdir(), 'prebuild.tar.gz')
+  var req = get(reqOpts, function (err, res) {
+    if (err) return onerror(err)
+    if (res.statusCode !== 200) return onerror()
+      pump(res, fs.createWriteStream(tempFile), function (err) {
+        var options = {
+          readable: true,
+          writable: true,
+          hardlinkAsFilesFallback: true
+        }
+        var binaryName
+        var updateName = function (entry) {
+          if (/\.node$/i.test(entry.name)) binaryName = entry.name
+        }
+        var targetFile = path.join(__dirname, 'builds', essential)
+        var extract = tfs.extract(targetFile, options)
+          .on('entry', updateName)
+        pump(fs.createReadStream(tempFile), zlib.createGunzip(), extract, function (err) {
+          if (err) return onerror(err)
+          cb()
+        })
+    })
+  })
+
+  req.setTimeout(30 * 1000, function () {
+    req.abort()
+  })
 }
 
-function deleteFolderRecursive(path, keepRoot) {
-  if( fs.existsSync(path) ) {
-    fs.readdirSync(path).forEach(function(file, index){
-      let curPath = path + "/" + file;
-      if(fs.lstatSync(curPath).isDirectory()) { // recurse
-        deleteFolderRecursive(curPath);
-      } else { // delete file
-        fs.unlinkSync(curPath);
-      }
-    });
-    if (!keepRoot) {
-      fs.rmdirSync(path);
-    }
+if (process.argv.indexOf('--all') > -1 || process.env.npm_config_targets) {
+  let chain = Promise.resolve()
+  let targets
+  if (process.env.npm_config_targets === 'all') {
```

Comment:
The end user / developer can choose what "prebuilds" to install when installing the package, we should probably set a better default here, and document how you can do this, I think you can alternatively add a "config" key to your package.json for providing the "npm_config_*" environment variables, that looks something like this:
```json
{
  "name": "my-package",
  "config": {
    "targets": "node-5.12.0,node-6.9.4",
    "platforms": "win32,darwin",
    "arches": "x64,ia32"
}
```

It's pretty handy when you want to get for example only windows builds or mac builds, or everything or just a single thing or... whatever you like really...